src/platform-aws.c

```diff
@@ -65,14 +69,26 @@ struct ec2_platform_data {
 		.gdr_required = true,
 		.net_flush_required = false,
 		.default_protocol = "RDMA",
+		.domain_per_endpoint = 0,
 	},
 	{
 		.name = "g5.48xlarge",
 		.topology = "g5.48xl-topo.xml",
 		.gdr_required = false,
 		.net_flush_required = true,
 		.default_protocol = "SENDRECV",
+		.domain_per_endpoint = 0,
 	},
+	{
+		.name = "trn1.32xlarge",
+		.default_protocol = "SENDRECV",
+		.domain_per_endpoint = 1,
+	},
+	{
+		.name = "trn1n.32xlarge",
+		.default_protocol = "SENDRECV",
+		.domain_per_endpoint = 1,
+	}
```

Comment:
I'm wondering whether we should set more fields here.
1. `gdr_required = true` and `net_flush_required = true` make sense here. Though I'm not sure if the latter (which influences setting the env var `NCCL_NET_FORCE_FLUSH`) will have any effect on Trainium.
2. It seems leaving latency set to 0 will cause us to return 0 as `props->latency` -- is this value relevant on Trainium? Incidentally, we leave it set to 0 on G5, which seems like a bug.

Reply:
latency is irrelevant on Trainium, and net_flush_required doesn't matter for us, but we can set it.